nibiru/query_clients/vpool.py

```diff
@@ -35,3 +38,11 @@ def base_asset_price(self, pair: str, direction: Direction, base_asset_amount: s
             pair=pair, direction=dir_pb, base_asset_amount=base_asset_amount
         )
         return self.query(self.api.BaseAssetPrice, req)
+
+
+def cast_str_to_float_safely(number_str: str) -> float:
+    try:
+        number = float(number_str)
+    except:
+        number = float(0)
```

Comment:
Don't we need numpy np.nan there?

I was doing this just now:
```python
        # The golang application is returning index price and twap price as string since v0.14.0
        # This is because those fields can be null, which leads to empty strings returned which cannot be easily
        # deserialized automatically by the deserialized function inside the self.query call above.
        # For this reason we do this manual update there

        for _, price in enumerate(resp["prices"]):

            for field in ["twap_mark", "index_price"]:
                value = price[field]
                if isinstance(value, str) and value == "":
                    value = np.nan
                else:
                    value = float(value)

                price[field] = value
```

Reply:
actually it's better to not have all the numpy dependency and reqs for the docker build we make

Reply:
I had the same idea. I wasn't sure if we should add the numpy dependency

Reply:
And both `""` and `None` are invalid inputs to the `float()` function, so it handles both cases